Fix small UI problems with the compose reply screen

alexjg opened the pull request:
Fixes #7781 

### Summary
Implements the small design tweaks in the linked issue

status: ready 

I've attached a screenshot of this running on a Pixel 2. I haven't tested on an iDevice.

![Screenshot (10 Sep 2019 00_24_36)](https://user-images.githubusercontent.com/224635/64573157-3589f480-d362-11e9-8c74-d74255ab5f22.png)

alexjg committed:
Fix small UI problems with the compose reply screen

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 2746b0d1 | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8959/1/) | 2019-09-09 23:41:01 | ~8 min | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8959/1/consoleText) |

errorists commented:
thanks @alexjg 
I'll take a more thorough look once the iOS build is ready, in the meantime judging by the attached screenshot the only issues I could find were:

• The reply icon is displayed at incorrect size (I'm guessing 20), it should be 16 by 16
• The close icon is using the incorrect asset, this is the right one. Not a huge difference, I wouldn't blame anyone for not spotting it :)
• Can we change the colour of the border at the top of the reply/chat toolbar to Light Grey, similar to how we use it elsewhere in the app?

<img width="931" alt="Screenshot 2019-09-10 at 10 15 29" src="https://user-images.githubusercontent.com/4932774/64596338-fc7a7000-d3b3-11e9-9d9b-e6a2d95fd583.png">

alexjg commented:
@errorists That asset looks like it's not been added to the app (that is, I can't see a `close-circle` asset). I can't seem to go the the master component from Figma (getting a 404). Where should I get the asset and are there any conventions I should be aware of when adding it to the iOS and Android projects?

errorists commented:
it's cool, I've [exported it here](https://www.dropbox.com/sh/olsrc85hnqnl4ca/AADvqI_LptPst3XlRsfQvgAEa?dl=0). As far as I can tell, we use PNGs, at 1x,2x,3x densities and tint it to get the correct colour. Alternatively, you can try and sample the close icon located under the Browser tab. It might be that it's made with a circular grey view and 'X' icon added on top, not sure, best to ask @flexsurfer 

<img width="735" alt="Screenshot 2019-09-10 at 16 08 04" src="https://user-images.githubusercontent.com/4932774/64621229-9c9cbd00-d3e5-11e9-841d-51731c00f2c1.png">

flexsurfer commented:
yes it's just a view with the icon, you can find it here https://github.com/status-im/status-react/blob/d07c9c6613551a6b68ff1087276f08aa64f61a71/src/status_im/ui/screens/browser/open_dapp/views.cljs#L49

alexjg commented:
Thanks for that, I've implemented it in a similar manner. I've also fixed the top border and the reply icon size. Let me know if anything else needs doing.

errorists commented:
looks good @alexjg ! 
@flexsurfer is there anything we can do to 'unstuck' the iOS build so it's ready to test?

flexsurfer commented:
@alexjg could you please rebase onto develop and squash commits, thanks

alexjg commented:
Done

yenda commented:
@alexjg linter is complaining about your recent changes, you can use `lein cljfmt fix` to fix it.

yenda commented on src/status_im/ui/screens/chat/styles/input/input.cljs:
```diff
@@ -13,49 +13,66 @@
   (assoc (message-author-name chosen?)
-         :padding-bottom 6))
+         :flex-shrink 1
+         :padding-left 0
```
does 0 padding and 0 margin change anything?

alexjg replied:
Yep, it's overriding the values from the definition of `message-author-name`, which is

```clojure
(defn message-author-name [chosen?]
  {:font-size           (if chosen? 13 12)
   :font-weight         (if chosen? "500" "400")
   :padding-top         6
   :padding-left        12
   :padding-right       16
   :margin-right        12
   :text-align-vertical :center
   :color               colors/gray})
```

alexjg commented:
> @alexjg linter is complaining about your recent changes, you can use `lein cljfmt fix` to fix it.

Done

flexsurfer commented:
@errorists ios ready to test

yenda commented:
@flexsurfer @errorists wasn't it tested already? can we merge?

errorists commented:
In the testing I just did, the reply icon is not displayed at all. Other than that I couldn’t find any issues

yenda commented:
@alexjg could you check the issue with the reply icons?

alexjg commented:
I'll check tomorrow, @errorists was the failure on any particular platform?

errorists commented:
Yes, iOS.

alexjg commented:
@errorists unfortunately I can't replicate this on android and I won't have access to an idevice until the week commencing 30th September. Is there any chance it's to do with asset naming? I don't completely understand how the asset loading works but I notice that there's a `tiny-reply.imageset` in the `ios` project whereas it's `tiny_reply.png` in the android assets.

errorists commented:
hey @alexjg . it's only a week, unless some savvy dev jumps here to help ( cc @yenda  ) I'd ask you to wait till then and try reproducing/fixing the image on iOS.

yenda commented on src/status_im/ui/screens/chat/input/input.cljs:
```diff
@@ -158,23 +158,29 @@
-     (chat-utils/format-reply-author from username current-public-key style/reply-message-author)
-     [react/text {:style (message-style/style-message-text false)} message-text]]))
+     [react/view {:style style/reply-message-to-container}
+      [vector-icons/tiny-icon :tiny-icons/tiny_reply {:container-style style/reply-icon
```
```suggestion
      [vector-icons/tiny-icon :tiny-icons/tiny-reply {:container-style style/reply-icon
```

yenda commented on src/status_im/ui/screens/chat/styles/input/input.cljs:
```diff
@@ -13,49 +13,66 @@
   (assoc (message-author-name chosen?)
-         :padding-bottom 6))
+         :flex-shrink 1
+         :padding-left 0
```
```suggestion
         ;; NOTE:  overriding the values from the definition of message-author-name
         :padding-left 0
```

yenda reviewed (dismissed):
No need to wait a week I'm pretty sure it is because of the underscore

yenda commented:
Hey @alexjg are you still planning on finishing this PR?

alexjg commented:
@yenda yep, I've been on holiday for the past few days, hence slow replies. I'll squash once I'm home tomorrow.

alexjg commented:
@yenda I've squashed and pushed.

One thing I've noticed is that the recent gfycat changes mean that the address of the account you're responding to appears rather than the gfycat of it. I'm not sure if this is a bug or not and it definitely seems unrelated to the functionality implemented here but I thought I would mention it.

Everything should be hunky dory for merge now.

yenda commented on src/status_im/ui/screens/chat/input/input.cljs:
```diff
@@ -158,23 +158,29 @@
+     [react/text {:style (assoc (message-style/style-message-text false) :font-size 14) :number-of-lines 3} message-text]]))
 
 (defview reply-message-view []
   (letsubs [{:keys [content from] :as message} [:chats/reply-message]]
```
```suggestion
  (letsubs [{:keys [content from alias] :as message} [:chats/reply-message]]
```

yenda commented on src/status_im/ui/screens/chat/input/input.cljs:
```diff
@@ -158,23 +158,29 @@
-        [react/view {:style style/cancel-reply-container}
-         [vector-icons/icon :main-icons/close {:container-style style/cancel-reply-icon
-                                               :color           colors/white}]]]])))
+        [reply-message from (:text content)]
```
```suggestion
        [reply-message from content]
```

yenda commented on src/status_im/ui/screens/chat/input/input.cljs:
```diff
@@ -158,23 +158,29 @@
   (letsubs [username           [:contacts/contact-name-by-identity from]
```
I can't make suggestions on this one since you didn't change it so here what it should look like:

```
(defview reply-message [alias {:keys [text name]]
```

churik commented:
@yenda can you review it again?
I'll check briefly after and we can merge

yenda commented:
@churik the code looks fine but the result isn't great because of the reply bug. I tried to fix it quickly but it is not that easy. As @alexjg mentions it comes from the username/identicons changes

churik commented:
@yenda 
In this case please just ping me here when I can test it again.

alexjg commented:
Is there anything I can do to move this forward? Or is it a case of waiting for a fix for the identicons issue which is being worked on elsewhere? I'm happy to have a look at the identicons issue myself if no one is doing so already.

yenda commented:
@alexjg can you include this small change in your PR https://github.com/status-im/status-react/pull/9076/files ? It is not the right fix because the alias will show as a username but at least it won't show a public key instead of the name

alexjg commented:
@yenda done

yenda commented on src/status_im/ui/screens/chat/utils.cljs:
```diff
@@ -12,22 +12,23 @@
-    :else
-    [react/text {:style {:color colors/gray :font-size 12 :font-weight "400"}}
-     from]))
+  (let [additional-styles (style false)]
```
sorry one last thing are you sure this is actually necessary? what is this argument for? I can't find anywhere where it is true, I'm wondering if we couldn't remove the whole `chosen?` param in these styles fns

alexjg replied:
Yeah it doesn't seem like it's used anywhere but I wasn't completely confident of that (given my inexperience with the codebase) so I made the smallest change I could. It certainly does seem like we could remove the `chosen?` parameter, want me to go ahead and do that?

yenda replied:
@flexsurfer

yenda replied:
I'm out this week ill let @flexsurfer decide

alexjg commented:
@yenda is there anything remaining to be done to get this merged? I'm keen to get it done as the gitcoin bounty attached to it expires in 2 days.

alexjg commented:
I've rebased from `develop` and the indenticon issue looks to be solved.

churik commented:
Tested on IOS 11.4.1. and Android 8 (LG V20):
- reply screen with long messages
- custom user pictures
- different types of chats

From my POV can be merged.
@flexsurfer can you review it please or we need to wait @yenda approve anyway?
